src/browser-pool.ts

```diff
@@ -704,6 +748,36 @@ export class BrowserPool<
             }, PAGE_CLOSE_KILL_TIMEOUT_MILLIS);
         }
     }
+
+    private _initializeFingerprinting(): void {
+        const { useFingerprintPerProxyCache = true, fingerprintPerProxyCacheSize = 10_000 } = this.fingerprintsOptions;
+        this.fingerprintGenerator = new FingerprintGenerator(this.fingerprintsOptions.fingerprintGeneratorOptions);
+        this.fingerprintInjector = new FingerprintInjector();
+
+        if (useFingerprintPerProxyCache) {
+            this.fingerprintCache = new FingerprintToProxyCache({ maxSize: fingerprintPerProxyCacheSize });
+        }
+
+        this._addFingerprintHooks();
+    }
+
+    private _addFingerprintHooks() {
+        this.preLaunchHooks = [
+            ...this.preLaunchHooks,
+            // This is flipped because of the fingerprint cache.
+            // It is usual to generate proxy per browser and we want to know the proxyUrl for the caching.
+            createFingerprintPreLaunchHook(this),
+
+        ];
+        this.prePageCreateHooks = [
+            createPrePageCreateHook(),
+            ...this.prePageCreateHooks,
+        ];
+        this.postPageCreateHooks = [
+            // @ts-expect-error nope it is not undefined because we initialize before we call this fce.
+            createPostPageCreateHook(this.fingerprintInjector),
```

Comment:
you can make it always defined via [non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator)

```suggestion
            createPostPageCreateHook(this.fingerprintInjector!),
```